packages/core/src/utilities/getScalarDataType.ts

```diff
@@ -0,0 +1,31 @@
+import { ScalingParameters } from '../types';
+
+/**
+ * If the scalar data is a Uint8Array, return 'Uint8Array'. If the scalar data is a
+ * Float32Array, return 'Float32Array'. If the scalar data is a Int16Array, return
+ * 'Int16Array'. If the scalar data is a Uint16Array, return 'Uint16Array'. If the
+ * scalar data is none of the above, throw an error.
+ * @param {ScalingParameters} scalingParameters - {
+ * @param {any} [scalarData] - The data to be converted.
+ * @returns The data type of the scalar data.
+ */
+export default function getScalarDataType(
```

Comment:
This needs to get expanded to consider scaling parameters to decide on the final data type (if scalings are float, then the `type` passed to wado as option should be float)

Reply:
I am assuming this is usually used for PET overlays. Would it be worth it to account for `preferSizeOverAccuracy` to perform a scale + cast